runtime/stdlib/test_test.go

```diff
@@ -651,6 +651,291 @@ func TestTestEqualMatcher(t *testing.T) {
 	})
 }
 
+func TestAssertEqual(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("success", func(t *testing.T) {
+		t.Parallel()
+
+		script := `
+		    import Test
+
+		    pub fun test() {
+		        Test.assertEqual("this string", "this string")
```

Comment:
In relation to my previous comment - Would something like Test.assertEqual(10, "this string") fail at compile time ?

If not it would be interesting to see a test like this in order to see if expected.Equal survives.